File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/consumer/offsets/ack/ProductListener.kt
package io.micronaut.kafka.docs.consumer.offsets.ack

import io.micronaut.configuration.kafka.annotation.KafkaListener
import io.micronaut.configuration.kafka.annotation.Topic
import io.micronaut.context.annotation.Requires
import io.micronaut.kafka.docs.Product
import io.micronaut.messaging.Acknowledgement

import io.micronaut.configuration.kafka.annotation.OffsetReset.EARLIEST
import io.micronaut.configuration.kafka.annotation.OffsetStrategy.DISABLED

@Requires(property = "spec.name", value = "AckProductListenerTest")
// tag::clazz[]
@KafkaListener(offsetReset = EARLIEST, offsetStrategy = DISABLED) // <1>
class ProductListener {

    @Topic("awesome-products")
    fun receive(product: Product, acknowledgement: Acknowledgement) { // <2>
        // process product record
        acknowledgement.ack() // <3>
    }
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/consumer/offsets/ack/ProductListener.py
from micronaut.configuration.kafka.annotation import KafkaListener, OffsetReset, OffsetStrategy, Topic
from micronaut.context.annotation import Requires
from micronaut.kafka.docs.Product import Product
from micronaut.messaging import Acknowledgement


@Requires(property="spec.name", value="AckProductListenerTest")
# tag::clazz[]
@KafkaListener(offsetReset=OffsetReset.EARLIEST, offsetStrategy=OffsetStrategy.DISABLED)  # <1>
class ProductListener:

    @Topic("awesome-products")
    def receive(self, product: Product, acknowledgement: Acknowledgement) -> None:  # <2>
        # process product record
        acknowledgement.ack()  # <3>
# end::clazz[]
